pandas/tests/generic/test_generic.py

```diff
@@ -872,6 +872,19 @@ def test_equals(self):
         assert a.equals(e)
         assert e.equals(f)
 
+    def test_equals_subclass(self):
+        # https://github.com/pandas-dev/pandas/pull/34402
+
+        class MySeries(pd.Series):
+            pass
+
```

Comment:
can you put this in tests/series/test_sublcass and similar for frame/test_sublcass

Reply:
This is not a test about the subclass working correctly (anything related to the subclassing machinery, eg correct use of `_constructor` to preserve the subclass in operations etc), but only about our own equals method accepting subclasses, so I think the test rather fits here with the other `equals` tests.

But I can certainly use the `tm.SubclassedSeries` helper to avoid creating a dummy class here.

Reply:
i think this belongs with the rest of the tests & needs for DataFrame subclass as well. having 2 places is just very friendly to future folks.

Reply:
Having 2 places where `equals` is tested is equally unfriendly I think .. 
Both ways have some aspect of the tests splitted in two places, and as mentioned above, this test is actually about our `equals` implementation, not about the behaviour of a subclassing mechanism, and thus more belongs here IMO.

Will add dataframe to the test

Reply:
pls just move this, ALL subclass tests are in the same file. This is breaking the pattern for no reason.